src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,446 @@
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Cold White",
+	})
+	public get supportsColdWhite(): boolean {
+		return this._supportsColdWhite;
+	}
+
+	private _supportsRed: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Red",
+	})
+	public get supportsRed(): boolean {
+		return this._supportsRed;
+	}
+
+	private _supportsGreen: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Green",
+	})
+	public get supportsGreen(): boolean {
+		return this._supportsGreen;
+	}
+
+	private _supportsBlue: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Blue",
+	})
+	public get supportsBlue(): boolean {
+		return this._supportsBlue;
+	}
+
+	private _supportsAmber: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Amber",
+	})
+	public get supportsAmber(): boolean {
+		return this._supportsAmber;
+	}
+
+	private _supportsCyan: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Cyan",
+	})
+	public get supportsCyan(): boolean {
+		return this._supportsCyan;
+	}
+
+	private _supportsPurple: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Purple",
+	})
+	public get supportsPurple(): boolean {
+		return this._supportsPurple;
+	}
+
+	private _supportsIndex: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Indexed Color",
+	})
+	public get supportsIndex(): boolean {
+		return this._supportsIndex;
+	}
+
+	public toJSON(): JSONObject {
+		return super.toJSONInherited({
+			supportsWarmWhite: this._supportsWarmWhite,
+			supportsColdWhite: this._supportsColdWhite,
+			supportsRed: this._supportsRed,
+			supportsGreen: this._supportsGreen,
+			supportsBlue: this._supportsBlue,
+			supportsAmber: this._supportsAmber,
+			supportsCyan: this._supportsCyan,
+			supportsPurple: this._supportsPurple,
+			supportsIndex: this._supportsIndex,
+		});
+	}
+}
+
+@CCCommand(ColorSwitchCommand.SupportedGet)
+@expectedCCResponse(ColorSwitchCCSupportedReport)
+export class ColorSwitchCCSupportedGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | CCCommandOptions,
+	) {
+		super(driver, options);
+	}
+}
+
+@CCCommand(ColorSwitchCommand.Report)
+export class ColorSwitchCCReport extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		// Docs say 'variable length', but the table shows 2 bytes.
+		validatePayload(this.payload.length >= 2);
+		this._colorComponent = this.payload[0];
+		this._value = this.payload[1];
+
+		// TODO: We probably want to store each color value
+		//	as a different ccValue.  How do we do that when
```

Comment:
You'll need to store the values manually in the value DB. `BinarySensorCCReport` has an example. I'll leave it up to you how to translate from the numeric color component to the property name.